torchrl/lightning/loops.py

```diff
@@ -0,0 +1,493 @@
+        lr: float = 3e-4,
+        max_grad_norm: float = 1.0,
+        frame_skip: int = 1,
+        frames_per_batch: int = 100,
+        total_frames: int = 100_000,
+        sub_batch_size: int = 1,
+        lr_monitor: str = "loss/train",
+        lr_monitor_strict: bool = False,
+        rollout_max_steps: int = 1000,
+        automatic_optimization: bool = True,
+        use_checkpoint_callback: bool = False,
+        save_every_n_train_steps: int = 100,
+        raise_error_on_nan: bool = False,
+        num_envs: int = 1,
+        env_kwargs: ty.Optional[ty.Dict[str, ty.Any]] = None,
+    ) -> None:
+        """
+        Args:
+            loss_module (TensorDictModule):
+                A torchl loss module.
+
+            policy_module (TensorDictModule):
+                Policy module.
+
+            value_module (TensorDictModule):
+                Value module.
+
+            target_net_updater (SoftUpdate, optional):
+                Target network updater. Defaults to None.
+
+            lr (float, optional):
+                Learning rate. Defaults to 3e-4.
+
+            max_grad_norm (float, optional):
+                Maximun norm value for the gradients. Defaults to 1.0.
+
+            frame_skip (int, optional):
+                Frame skip value. Defaults to 1.
+
+            frames_per_batch (int, optional):
+                Number of frames per batch. Defaults to 100.
+
+            total_frames (int, optional):
+                Total number of frames. Defaults to 100_000.
+
+            sub_batch_size (int, optional):
+                Sub batch size. Defaults to 1.
+
+            lr_monitor (str, optional):
+                Name given to the loss to monitor.
+                Defaults to "loss/train".
+
+            lr_monitor_strict (bool, optional):
+                Whetehr to raise an error in case the metric `lr_monitor` is not found.
+                Defaults to False.
+
+            rollout_max_steps (int, optional):
+                Max number of rollout steps.
+                Defaults to 1000.
+
+            automatic_optimization (bool, optional):
+                Whether to use manual or automatic optimization.
+                Defaults to True.
+
+            use_checkpoint_callback (bool, optional):
+                Whether to use a checkpoint callback.
+                Defaults to False.
+
+            save_every_n_train_steps (int, optional):
+                Save every n steps. Defaults to 100.
+
+            raise_error_on_nan (bool, optional):
+                Whether to immediately raise an error if a NaN is encountered.
+                Defaults to False.
+
+            num_envs (int, optional):
+                Number of environments to use.
+                If `num_envs > 1`, `num_envs` copies of the input env will be created.
+                Defaults to 1.
+
+            env_kwargs (ty.Dict[str, ty.Any], optional):
+                Parameters for your environment.
+                Defaults to {}.
```

Comment:
```suggestion
            loss_module (TensorDictModule): A torchl :class:`~torchrl.objectives.LossModule` module.
            policy_module (TensorDictModule): Policy module.
            value_module (TensorDictModule): Value module.
            target_net_updater (SoftUpdate, optional): Target network updater. Defaults to ``None``.
            lr (float, optional): Learning rate. Defaults to ``3e-4``.
            max_grad_norm (float, optional): Maximum norm value for the gradients. Defaults to ``1.0``.
            frame_skip (int, optional): Frame skip value. Defaults to 1.
            frames_per_batch (int, optional): Number of frames per batch. Defaults to ``100``.
            total_frames (int, optional): Total number of frames. Defaults to ``100_000``.
            sub_batch_size (int, optional): Sub batch size. Defaults to ``1``.
            lr_monitor (str, optional): Name given to the loss to monitor.
                Defaults to ``"loss/train"``.
            lr_monitor_strict (bool, optional):
                Whether to raise an error in case the metric `lr_monitor` is not found.
                Defaults to ``False``.
            rollout_max_steps (int, optional): Max number of rollout steps.
                Defaults to ``1000``.
            automatic_optimization (bool, optional): Whether to use manual or automatic optimization.
                Defaults to ``True``.
            use_checkpoint_callback (bool, optional): Whether to use a checkpoint callback.
                Defaults to ``False``.
            save_every_n_train_steps (int, optional): Save every n steps. Defaults to ``100``.
            raise_error_on_nan (bool, optional): Whether to immediately raise an error if a NaN is encountered.
                Defaults to ``False``.
            num_envs (int, optional): Number of environments to use.
                If `num_envs > 1`, `num_envs` copies of the input env will be created.
                Defaults to ``1``.
            env_kwargs (ty.Dict[str, ty.Any], optional): Parameters for your environment.
                Defaults to ``{}``.
```